diff/parse.go

```diff
@@ -136,9 +136,75 @@ func parseFileHeader(line string) (filename, timestamp string) {
 	ss := line[len(tokenOldFile)+1:]
 	tabi := strings.LastIndex(ss, "\t")
 	if tabi == -1 {
-		return ss, ""
+		return unescapeCString(ss), ""
 	}
-	return ss[:tabi], ss[tabi+1:]
+	return unescapeCString(ss[:tabi]), ss[tabi+1:]
+}
+
+func unescapeCString(str string) string {
```

Comment:
Can you add unit tests for this function and add some test cases (e.g. test file name which contains `\t`)